### PR TITLE
Issue 3264: Return a SegmentHandle from Storage create method.

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
Segment create returns a read-write SegmentHandle . This  eliminates the redundant fencing/ checks.

**Purpose of the change**  
Fixes [#3264](https://github.com/pravega/pravega/issues/3264)

**What the code does**  
All SyncStorage implementations now return  read-write SegmentHandle (and avoid redundant checks/file access)
 
**How to verify it**  
All unit tests and integration tests should pass.
Some improvement in performance.

### CODE REVIEW

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/attributes/SegmentAttributeBTreeIndex.java [hunk 1]
```diff
@@ -160,7 +160,8 @@
                         this.storage.openWrite(attributeSegmentName).thenAccept(this.handle::set),
                         ex -> ex instanceof StreamSegmentNotExistsException,
                         () -> this.storage.create(attributeSegmentName, this.config.getAttributeSegmentRollingPolicy(), timer.getRemaining())
-                                          .thenComposeAsync(si -> this.storage.openWrite(attributeSegmentName).thenAccept(this.handle::set), this.executor))
+                                          .thenComposeAsync(si -> this.storage.openWrite(attributeSegmentName)
```
Comment (andreipaduroiu) [inline segmentstore/server/src/main/java/io/pravega/segmentstore/server/attributes/SegmentAttributeBTreeIndex.java]: We can get rid of this line. `create` returns a handle and that's all we need.

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java [hunk 2]
```diff
@@ -279,8 +279,9 @@ public SegmentProperties create(String segmentName, SegmentRollingPolicy rolling
             throw ex;
         }
 
+        val handle = openHandle(segmentName, false);
```
Comment (andreipaduroiu) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java]: Look at line 264. Save the `new RollingSegmentHandle` in a variable and return that. It should contain the same info without having to read the data we've just written.

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/attributes/SegmentAttributeBTreeIndex.java (hunk 1 above)
Comment (sachin-j-joshi) [reply segmentstore/server/src/main/java/io/pravega/segmentstore/server/attributes/SegmentAttributeBTreeIndex.java]: oops. I missed that. will fix.

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java (hunk 2 above)
Comment (sachin-j-joshi) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java]: I tried that before , however that generates java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException.   So there is a little bit extra that needs to happen before returning. I'm working on fixing this.

**Pull request merged**